openBCIBoard.js

```diff
@@ -1,14 +1,471 @@
+        //    console.log('Whoops... ' + error);
+        //    return Promise.resolve();
+        //}).catch(function(err) {
+        //    self.badPackets++;
+        //    console.log('Dropped a packet :( -> Total Packets Dropped: ' + self.badPackets);
+        //    return Promise.resolve(err);
+        //});
+    };
+
+    /**
+     * Call to stop the stream
+     */
+    OpenBCIBoard.prototype.streamStop = function() {
+        this.streaming = false;
+        writeAndDrain(this.serial,k.OBCIStreamStop);
+    };
+    /**
+     * Call to disable the 60Hz line filter
+     */
+    OpenBCIBoard.prototype.filterDisable = function() {
+        if(this.connected) {
+            writeAndDrain(this.serial, k.OBCIFilterDisable);
+        }
+    };
+    /**
+     * Call to enable the 60Hz line filter
+     */
+    OpenBCIBoard.prototype.filterEnable = function() {
+        if(this.connected) {
+            writeAndDrain(this.serial, k.OBCIFilterEnable);
+        }
+    };
+
+    OpenBCIBoard.prototype.sampleRate = function() {
+        if(this.daisy) {
+            return k.OBCISampleRate125;
+        } else {
+            return k.OBCISampleRate250;
+        }
+    };
+
+    OpenBCIBoard.prototype.numberOfChannels = function() {
+        if(this.daisy) {
+            return k.OBCINumberOfChannelsDaisy;
+        } else {
+            return k.OBCINumberOfChannelsDefault;
+        }
+    };
+
+    OpenBCIBoard.prototype.processBytes = function(data) {
+        var self = this;
+        var sizeOfData = data.byteLength;
+        if(self.lookingForMoney) { //in a reset state
+            for (var i = 0; i < sizeOfData - 2; i++) {
+                if (self.moneyBuf.equals(data.slice(i, i + 3))) {
+                    console.log('Money!');
+                    self.lookingForMoney = false;
+                    self.emit('ready');
+                }
+            }
+        } else { //ready to open the serial fire hose
+            // send input data to master buffer
+            self.bufMerger(self,data);
+
+            // parse the master buffer
+            while(self.masterBuffer.packetsRead < self.masterBuffer.packetsIn) {
+                var rawPacket = self.bufPacketStripper(self);
+                var newSample = OpenBCISample.convertPacketToSample(rawPacket);
+                if(newSample) {
+                    //console.log('Wow, for the first time, you actually got a packet... maybe lets check it out!');
+                    self.emit('sample', sample);
+                } else {
+                    self.badPackets++;
+                    /*TODO: initiate messed up packet protocol, need to sync back up with good packet */
+                }
+                OpenBCISample.convertPacketToSample(rawPacket).then(function(sample) {
+
+                }, function(err) {
+
+                });
+
+            }
+        }
+    };
+
+    OpenBCIBoard.prototype.autoFindOpenBCIBoard = function(callback) {
+        var macSerialPrefix = 'usbserial-D';
```

Comment:
i think it can also show up `tty-*`. we can make a list of common prefixes (should be 2 or 3 max) and check those for the autoFind